### PR TITLE
Feature: Warn when attempting to send tx with data to non-contract

### PR DESCRIPTION
**Pull request by kumavis**
Modified version of https://github.com/MetaMask/metamask-extension/pull/5283

`tx-gas-util` and `estimateGas` have evolved and will later need to be renamed to more correctly reflect their role

### CODE REVIEW

**Comment by kumavis** [comment]
This is crashing because the transaction `origin` data is missing - not sure why

**Comment by kumavis** [comment]
This is failing because the old ui does not handle an error being thrown here

**Comment by HowardBraham** [comment]
@kumavis Do you need any help here?  What can I do to assist?

File: old-ui/app/components/pending-tx.js [hunk 1]
```diff
@@ -19,7 +20,9 @@ const BNInput = require('./bn-as-decimal-input')
 const MIN_GAS_PRICE_BN = new BN('0')
 const MIN_GAS_LIMIT_BN = new BN('21000')
 
-module.exports = PendingTx
+module.exports = connect()(PendingTx)
```
Comment (alextsg) [inline old-ui/app/components/pending-tx.js]: Don't think this `connect()` is needed

**Review by alextsg — APPROVED** [review]

File: old-ui/app/components/pending-tx.js (hunk 1 above)
Comment (kumavis) [reply old-ui/app/components/pending-tx.js]: it is because `dispatch` is called

**Pull request merged**